components/feature/prompts/src/test/java/mozilla/components/feature/prompts/creditcard/CreditCardPickerTest.kt

```diff
@@ -105,6 +112,32 @@ class CreditCardPickerTest {
         verify(creditCardSelectBar).showPrompt(promptRequest.creditCards)
     }
 
+    @Test
+    fun `GIVEN a selected credit card WHEN onAuthSuccess is called THEN the confirmed credit card is received`() {
+        assertNull(creditCardPicker.selectedCreditCard)
+
+        setupSessionState(promptRequest)
+
+        creditCardPicker.onOptionSelect(creditCard)
+        creditCardPicker.onAuthSuccess()
+
+        assertEquals(creditCard, confirmedCreditCard)
+        assertNull(creditCardPicker.selectedCreditCard)
+    }
+
+    @Test
+    fun `GIVEN a selected credit card WHEN onAuthFailuree is called THEN the prompt request is dismissed`() {
```

Comment:
Could we also confirm the prompt request was consume and dismissed? :)

Reply:
Yes, we already check with `assertTrue(onDismissCalled)` below

Reply:
Thanks for the clarification, as this value is shared for multiple tests, it could be reset it on `setup()` 👍🏽 .
Maybe making it default to null could be a good idea.